api/users/views.py

```diff
@@ -896,7 +896,7 @@ def get(self, request, *args, **kwargs):
     def perform_destroy(self, instance):
         user = self.get_user()
         email = instance.address
-        if instance.confirmed:
+        if instance.confirmed and instance.verified:
```

Comment:
test please!

Reply:
☝️

Probably add to:
https://github.com/CenterForOpenScience/osf.io/blob/398285e8c158786c3e79ee1f8aaaefbc4f6a5cd8/api_tests/users/views/test_user_settings.py#L472-L494